<launch>

	<!-- ar marker detection server -->
	<arg name="markerId" default="26"/>
	<arg name="markerSize" default="0.07"/>
	<arg name="camera_info" default="/camera_info"/>
	<arg name="image_topic" default="/image"/>
	<include file="$(find o2as_aruco_marker_detection)/launch/marker_detection.launch">
		<arg name="markerId" value="$(arg markerId)"/>
		<arg name="markerSize" value="$(arg markerSize)"/>
		<arg name="camera_info" value="$(arg camera_info)"/>
		<arg name="image_topic" value="$(arg image_topic)"/>
	</include>

	<!-- marker pose estimation -->
	<arg name="camera_name" default=""/>
	<arg name="camera_type" default=""/>
	<arg name="camera_serial_number" default=""/>
	<node name="marker_detection" pkg="o2as_aruco_marker_detection" type="marker_detection_demo.py" output="screen">
		<param name="marker_id" type="int" value="26"/>
		<param name="image_dir" type="string" value="$(find o2as_aruco_marker_detection)/data/image"/>
		<param name="camera_name" type="string" value="$(arg camera_name)"/>
		<param name="camera_type" type="string" value="$(arg camera_type)"/>
	</node>

	<!-- <node name="relative_pose" pkg="o2as_aruco_marker_detection" type="relative_pose.py" output="screen"/> -->

	<!-- reconfigureation -->
	<node name="reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure"/>

	<!-- view -->
	<node name="image_view" pkg="image_view" type="image_view" args="image:=$(arg image_topic)"/>
	<node name="debug_view" pkg="image_view" type="image_view" args="image:=/marker_detector/debug"/>
	<node name="result_view" pkg="image_view" type="image_view" args="image:=/marker_detector/result"/>

</launch>
